storage/local/persistence.go

```diff
@@ -114,10 +121,49 @@ type persistence struct {
 
 // newPersistence returns a newly allocated persistence backed by local disk storage, ready to use.
 func newPersistence(basePath string, chunkLen int, dirty bool) (*persistence, error) {
-	if err := os.MkdirAll(basePath, 0700); err != nil {
+	dirtyPath := filepath.Join(basePath, dirtyFileName)
+	versionPath := filepath.Join(basePath, versionFileName)
+
+	if file, err := os.Open(versionPath); err == nil {
+		defer file.Close()
+		data := make([]byte, 8)
+		n, err := file.Read(data)
```

Comment:
Ah, right. I was looking for that and couldn't find it. Same for ReadDir. Will incorporate those.